src/hooks/useOnScrollReachedEndDetector/index.ts

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from 'react';
+import { SCROLL_BUFFER } from '../../utils/consts';
+import { isAboutSame } from '../../modules/Channel/context/utils';
+import { useDebounce } from '../useDebounce';
+
+const BUFFER_DELAY = 500;
+
+export interface UseOnScrollReachedEndDetectorProps {
+  scrollRef: React.RefObject<HTMLDivElement>;
+  onReachedTop?: () => void;
+  onReachedBottom?: () => void;
+  onInBetween?: () => void;
+}
+
+export function useOnScrollPositionChangeDetector(props: UseOnScrollReachedEndDetectorProps): () => void {
+  const {
+    scrollRef,
```

Comment:
Is it possible to use event from onScroll instead of scollRef?

Reply:
We could but im not sure how to change dependency from current:
```
[scrollRef, onReachedTop, onReachedBottom, onInBetween]
```
Could you enlight me please?

Reply:
If you switch to usePreservedCallback, it seems like the dependency array will no longer be necessary

Reply:
ah then lets resolve this bc usePreservedCallback has been applied

Reply:
Could we change to event instead of ref? Receiving values related to scrolling using a ref seems a bit awkward because it requires a separate execution of callbacks within the event handler.

To ensure consistency between the ref of the element and the actual element triggering the event, it appears more favorable to eliminate the ref and utilize the event directly.

Reply:
Done!